PIL/PalmImagePlugin.py

```diff
@@ -80,13 +80,10 @@
 
 # so build a prototype image to be used for palette resampling
 def build_prototype_image():
-    image = Image.new("L", (1, len(_Palm8BitColormapValues),))
+    image = Image.new("L", (1, len(_Palm8BitColormapValues)))
     image.putdata(list(range(len(_Palm8BitColormapValues))))
-    palettedata = ()
-    for colormapValue in _Palm8BitColormapValues:
-        palettedata += colormapValue
-    for i in range(256 - len(_Palm8BitColormapValues)):
-        palettedata += (0, 0, 0)
+    palettedata = sum(_Palm8BitColormapValues, ()) +\
```

Comment:
Looking at that sum, I think it's an optimization/obscuration too far. I think it's just doing the same thing as before, but it's all hidden behind a badly named function.  It's still making N tuples, each one longer than the last.  But, it's only <=256, so it's not that bad.